editor/reducer.js

```diff
@@ -521,6 +521,45 @@ export function notices( state = {}, action ) {
 	return state;
 }
 
+const defaultMetaboxState = {
+	advanced: {
+		isActive: false,
+		isDirty: false,
+		isUpdating: false,
+	},
+	normal: {
+		isActive: false,
+		isDirty: false,
+		isUpdating: false,
+	},
+	side: {
+		isActive: false,
+		isDirty: false,
+		isUpdating: false,
+	},
+};
+
+export function legacyMetaboxes( state = defaultMetaboxState, action ) {
+	switch ( action.type ) {
+		case 'INITIALIZE_METABOX_STATE':
+			const newState = {};
+
+			for ( const location in action.metaboxes ) {
+				newState[ location ] = { ...state[ location ], isActive: action.metaboxes[ location ] };
+			}
+
+			return newState;
+		case 'HANDLE_METABOX_RELOAD':
+			return { ...state, [ action.location ]: { ...state[ action.location ], isUpdating: false, isDirty: false } };
+		case 'REQUEST_METABOX_UPDATE':
+			return { ...state, [ action.location ]: { ...state[ action.location ], isUpdating: true, isDirty: false } };
```

Comment:
Is it really fair to say that the post is not dirty at this point, since the save is still pending completion? How does this tie into the page dirty prompt behavior?

Reply:
Good point. If the post save is unsuccessful, this would be a problem. I will have to think about how to fix this.

Reply:
So granted this is confusing, but the request for metabox updating does not happen until the post has successfully saved, while that is updating, we want to keep the state not dirty. Visually it looks like this happens from when you hit `Update/Publish`, but that is not what is actually happening state wise. Potentially what should be done is to have an effect on `REQUEST_METABOX_UPDATES` that will keep the post save button in a state of updating?